src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -14,7 +15,11 @@ internal abstract partial class BoundExpression
         /// </summary>
         public virtual object Display
         {
-            get { return this.Type; }
+            get
+            {
+                Debug.Assert((object)this.Type != null, $"Unexpected null type in {this.GetType().Name}");
```

Comment:
null is never a good argument for a diagnostics. While it is good to have an assert, I think we should also add a fallback to something non-null.